### PR TITLE
refactor: Unified single response card and improved UX

### LINKED ISSUE #454: [REFACTOR] Unify SingleResponse Card and increase UX
**Is your feature request related to a problem? Please describe.**
Currently, the single response card on the Person Detail page differs from the single response card in the Responses tab in the survey summary. Furthermore, skipped questions are not displayed in a user-friendly way and the tagging functionality is missing on the Person Detail page.

**Describe the solution you'd like**
- Rework the data handling to the new Services
- Create one, reusable component
- Update the UX as suggested below
- Use [Server Components](https://nextjs.org/docs/getting-started/react-essentials#server-components) & [Server Actions](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions). Use Client Components only when necessary.

## Rework data handling
We're moving to Services and we need a new Service that returns all responses per user called "getPersonResponses". You find "getSurveyResponses" in "package/lib/services/response.ts" which you can copy and adapt.
Please use a server action to mutate data (when creating a new responseNote).

## Rework data passed to component
Currently, the SingleResponse is passed a `data` object, a survey and an environmentId. It needs to be rewritten so that we can pass a `response` and a `survey` to the component for it to display the data. The `survey` object contains the questions and the environmentId for linking and the `response` object the answers and person details.

## Rework UI
Here is an overview of the elements the Single Response Card needs on the Response Tab:

grafik

- Respondent (id or email) or Anonymous (see SingleResponse.tsx)
- Date
- Delete icon
- Notes
- Tags

Here is an overview of the elements the Single Response Card needs on the Person Detail page:

grafik

- Survey Name (linked to survey summary)
- Date
- Delete icon
- Tags

Make sure to design the component accordingly with the necessary props.

For displaying the questions, there are several cases to consider. Here is a list:

1. Survey completed, all questions answered:
grafik

2. Survey completed, one or some questions skipped in the middle. Please note that only **if there are two or more questions** skipped in a row, the icon will appear. It is the `chevrons-right` icon from the lucide icons package:
grafik
grafik

3. Survey completed and last question skipped:
grafik

5. Survey closed after filling question:
grafik

7. Survey aborted after skipping question:
grafik

Please note that the `border-l` goes from `dashed` if the question is skipped to `dotted` if the survey was closed before

### How we code at Formbricks 🤓
- Everything is type-safe
- All UI components are in the package `formbricks/ui`
- Run `pnpm dev` to find a demo app to test in-app surveys at `localhost:3002`
- We use **chatGPT** to help refactor code. Use our [Formbricks ✨ megaprompt ✨](https://github.com/formbricks/formbricks/blob/main/megaprompt.md) to create the right context before you write your prompt.

### PR DESCRIPTION
**Pull request by Dhruwang**
## What does this PR do?

Unified single response card and improved UX

Fixes #454 

Single response card for peoples page looks like this :

Screenshot from 2023-08-30 19-33-20

Also improved UI for displaying user responses

Screenshot from 2023-08-30 20-14-27

Screenshot from 2023-08-29 19-19-07

Screenshot from 2023-08-30 20-14-54

Screenshot from 2023-08-30 20-15-08

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

Check single response card

## Checklist

- [x] Added a screen recording or screenshots to this PR
- [x] Filled out the "How to test" section in this PR
- [x] Read the [contributing guide](https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [x] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

**Comment by jobenjada** [comment]
Hey @Dhruwang thanks a lot, looks really good!!

I spotted some bugs and have further suggestions :)

**Rating question type not displayed**
image

Make sure it works for stars and emojis as well:
image


**Pls display userId as main identifier also in Person Detail Page and Summary view**

Priorities: Email -> UserId -> Internal Formbricks userId

Pls make a reusable component for this to keep our code DRY

image

image

**Multi Select Questions cannot be skipped**
When a user does not select one option in a multi select question it is not marked as 'skipped'
image

**Not related to this ticket but you're the right man to fix it :)**
When toggling between Summary and Reponse tab, I want all data to be pulled from the server. I had the issue that I had a response in the Summary tab which wasnt in the Response tab. Had to reload whole page:
image

Good work, really like the look and feel of it! 😍

**Comment by jobenjada** [comment]
@Dhruwang please resolve merge conflicts for this to get merged :)

**Review by mattinannt — COMMENTED** [review]
@Dhruwang Thanks a lot for the changes. Looks great :-) I would like to use this PR as a test for the new component structure as we here face the problem of reusable components. Can you please make the changes as stated and also discussed in Slack and see if this is working? :-)

File: packages/ui/Smileys.tsx
Comment (mattinannt) [inline packages/ui/Smileys.tsx:1]: Please also move to it's own directory `Smileys`

File: packages/ui/index.tsx
Comment (mattinannt) [inline packages/ui/index.tsx:1]: Please also export the new components DeleteDialog, Modal, Smileys and SurveyStatusIndicator

File: packages/ui/Smileys.tsx
Comment (mattinannt) [reply packages/ui/Smileys.tsx:1]: Make sure to delete the old `Smileys.tsx` in `apps/web/components` and reference the file from `@formbricks/ui` where used.

File: packages/ui/DeleteDialog/index.tsx
Comment (mattinannt) [inline packages/ui/DeleteDialog/index.tsx:1]: Make sure to delete the old `DeleteDialog.tsx` in `apps/web/components` and reference the file from `@formbricks/ui` where used.

File: packages/ui/Modal/index.tsx
Comment (mattinannt) [inline packages/ui/Modal/index.tsx:1]: Make sure to delete the old `Modal.tsx` in `apps/web/components/shared` and reference the file from `@formbricks/ui` where used.

**Review by mattinannt — COMMENTED** [review]
@Dhruwang Thank you. The new structure in `@formbricks/ui` looks great. Please make sure to not have the components in the codebase twice and delete the old components. I started to write PR comments for this but it seems that's true for all components you newly created in `@formbricks/ui`

**Review by mattinannt — APPROVED** [review]

**Pull request merged**